content/source/docs/index.html.markdown

```diff
@@ -1,14 +1,95 @@
 ---
-layout: "docs"
+layout: "docs-frontpage"
 page_title: "Documentation"
-sidebar_current: "docs-home"
+sidebar_current: "docs-frontpage"
 description: |-
-  Welcome to the Terraform documentation! This documentation is more of a reference guide for all available features and options of Terraform. If you're just getting started with Terraform, please start with the introduction and getting started guide instead.
+  A brief map of the documentation for Terraform CLI, Terraform Enterprise, the
+  Terraform GitHub Actions, and the rest of the Terraform ecosystem.
 ---
 
 # Terraform Documentation
 
-Welcome to the Terraform documentation! This documentation is more of a reference
-guide for all available features and options of Terraform. If you're just getting
-started with Terraform, please start with the
-[introduction and getting started guide](/intro/index.html) instead.
+Welcome to the Terraform documentation!
+
+There are a lot of things a user of Terraform might need to know about. To help you manage this information, we've divided the Terraform docs into several sections.
+
+You can navigate the docs via this index page, or you can jump from section to section using the "Other Docs" area of the navigation sidebar, available in most areas of this site.
+
+<div class="container-fluid"><div class="row">
+
+<div class="col-md-6 col-sm-12">
+
+### [Terraform Glossary](/docs/glossary.html)
+
+Definitions (and helpful links) for technical terms used throughout Terraform's documentation, help text, and UI. Visit the glossary whenever you get lost.
+
+### [Terraform CLI](/docs/cli-index.html)
+
+-> Intermediate and advanced users spend most of their time here.
+
+Documentation for Terraform's core functionality, including:
+
+- [Terraform's configuration language](/docs/configuration/index.html)
+- [The `terraform` binary and its subcommands](/docs/commands/index.html)
+- [The main Terraform providers](/docs/providers/index.html)
+
+...and much more.
+
+### [Terraform Enterprise](/docs/enterprise/index.html)
+
+Documentation for Terraform Enterprise.
+
+Terraform Enterprise is a supplementary paid product that makes it easier for teams to manage infrastructure together. It provides a way to share access to sensitive data, a consistent and reliable environment for Terraform runs, access controls for approving changes to infrastructure, a private registry for sharing Terraform modules, detailed policy controls for governing the contents of Terraform configurations, and more.
+
+### [Introduction to Terraform](/intro/index.html)
```

Comment:
Given that this section of the page is visually relatively small and aimed at less-familiar users, I wonder about putting it at the top of this column instead of the bottom. That way we can redirect new users to where they need to be without detracting _too_ much from the navigation experience for those who already saw it.

Reply:
Yeah, might as well!